practice-app/server/src/api/controllers/lesson/get_lesson_by_lecturer.js

```diff
@@ -1,12 +1,18 @@
 import mongoose from 'mongoose';
 import { Lesson } from '../../../models/index.js';
+import { User } from '../../../models/index.js';
 
 export default async (req, res) => {
   const lecturerId = req.query.lecturer;
   if (!lecturerId || !mongoose.isValidObjectId(lecturerId)) {
     return res.status(400).json({ "resultMessage": "Please provide a valid lecturer id." });
   }
 
+  const lecturer = await User.findOne({ _id: lecturerId})
+  .catch((err) => {
+      return res.status(404).json({ "resultMessage": err.message });;
```

Comment:
This part represents an internal error, so its error status code should be 500 if I'm not wrong.